### DISCUSSION TITLE
Compatibility with Sinocare Sensor

### DISCUSSION THREAD

**Discussion by UberQueiroz** [NightscoutFoundation/xDrip#3393] [General]
Hi! Sinocare has one of the most low profile Bluetooth sensor amongst all Chinese manufacturers and I would like to suggest that xDrip would allow us to use Sinocare's with it, to make it compatible. Thanks!

![20240318_002506.jpg](https://github.com/NightscoutFoundation/xDrip/assets/66580535/d5d77ce1-f3cc-4574-96dd-1df3b426b59f)
)

**Comment by Navid200** [comment]
Are you in contact with the manufacturer who is interested in collaboration?

**Comment by mtumashenko** [comment]
No update for Sinocare iCan sensors or iCan Health app on Android?

**Reply by Navid200** [reply]
Does the Android app show a notification that includes readings?

**Reply by PhilTU** [reply]
> Does the Android app show a notification that includes readings?

apparently yes, I've asked for a screenshot (FB) ..... second notification (xDrip is on a Dexcom 1) :- 

![488041149_23886750184255910_8103482255028779199_n](https://github.com/user-attachments/assets/9bef9c53-ef38-4900-a42c-7bb885e5cc59)

**Comment by Navid200** [comment]
The iCan app is added to the list of apps xDrip can read from as a companion app as of the 2025.04.03 release.

**Reply by PhilTU** [reply]
Thanks for the prompt response

**Reply by mtumashenko** [reply]
![image](https://github.com/user-attachments/assets/5966dab4-20e2-4cc6-a54b-405797c72c18)
Hi! Thanks for help. I tried to get Xdrip working with iCan app using the latest nightly build, but with no luck. Selected Companion app source and gave permissions, but have no data in Xdrip.
Any suggestions?

**Reply by Navid200** [reply]
@mtumashenko Post a screenshot of the xDrip classic status page.

**Reply by PhilTU** [reply]
@mtumashenko does xDrip+ have permission to *read* notifications - android settings https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

**Reply by mtumashenko** [reply]
> @mtumashenko Post a screenshot of the xDrip classic status page.

![image](https://github.com/user-attachments/assets/1a0b34b1-5c4a-4a6d-ba27-3b59ffbb8510)

**Reply by mtumashenko** [reply]
> @mtumashenko does xDrip+ have permission to _read_ notifications - android settings https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

Yeah, first time I just granted permissions when installed xDrip and selected companion mode. Then I searched for accessibility settings and found that xDrip was showing as OFF and greyed out (can't change with "Restricted settings" error). After that I checked once more Apps - Settings menu and found that Access restricted permissions is not granted for xDrip (but I sure it was). So I granted restricted permissions, checked Accessibility - xDrip is now ON.
Nothing changed though, no data in xDrip from iCan app. And I can't find permission settings with exactly these names "Read notifications" and "Full control". The most similar setting I have in my daughter's Samsung with Android 14 in "Apps - Special Access" is Access Notifications, and it is already granted for xDrip:
![image](https://github.com/user-attachments/assets/bfcb55eb-d228-48df-ac82-581b62cad225)

**Reply by Navid200** [reply]
![430500417-bfcb55eb-d228-48df-ac82-581b62cad225](https://github.com/user-attachments/assets/51eda8e1-d2c8-44c7-8332-4186a656a124)

**Reply by PhilTU** [reply]
> Access Notifications, and it is already granted for xDrip

Ok !

**Reply by Cyxapuk-git** [reply]
I found that it may be different app's id https://github.com/NightscoutFoundation/xDrip/pull/3960

**Reply by mtumashenko** [reply]
Answered in that conversation too, package name installed on my daughter's phone:
https://play.google.com/store/apps/details?id=com.sinocare.ican.health.ce

**Reply by mtumashenko** [reply]
> ![430500417-bfcb55eb-d228-48df-ac82-581b62cad225](https://private-user-images.githubusercontent.com/51497406/430507806-51eda8e1-d2c8-44c7-8332-4186a656a124.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwMTQ3NTksIm5iZiI6MTc0NDAxNDQ1OSwicGF0aCI6Ii81MTQ5NzQwNi80MzA1MDc4MDYtNTFlZGE4ZTEtZDJjOC00NGM3LTgzMzItNDE4NmE2NTZhMTI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA3VDA4MjczOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjODEzYjNlNGNkMjhkYmE5YmE5ZjI0NGJmNjBkYjQxNDMyMGQyZDdkNmRlYzk2ZTMzMGZjYjM4YWM2MzY1YmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gRFQSAMtvtVjJm-K7YFCmskDQTi0Pli9Ukp06gcA9nI)

This app sends 2 different types of notifications, BG level and alarms

**Reply by mtumashenko** [reply]
> ![430500417-bfcb55eb-d228-48df-ac82-581b62cad225](https://private-user-images.githubusercontent.com/51497406/430507806-51eda8e1-d2c8-44c7-8332-4186a656a124.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQxMTMxMzEsIm5iZiI6MTc0NDExMjgzMSwicGF0aCI6Ii81MTQ5NzQwNi80MzA1MDc4MDYtNTFlZGE4ZTEtZDJjOC00NGM3LTgzMzItNDE4NmE2NTZhMTI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA4VDExNDcxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY3NzZmMGE0N2VkZjcxMDRhNjJkZDBjOWJiYjA0NGYyYTk4Y2ZhOWZiMzY3ODNhNjUwNDAyMjk3ZThmNDk4ZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LBcW--4rdxDvqdnjpQfITbtpI_dFvR0bphpv3s1J40c)

This is what it looks like in general and expanded (don't know if it does matter):

![image](https://github.com/user-attachments/assets/69fee949-15d8-4b44-94b5-91f3c7a4052c)

![image](https://github.com/user-attachments/assets/e07e165d-9156-411c-9cc4-995560014416)

**Comment by Navid200** [comment]
@Cyxapuk-git 

> I found that it may be different app's id  

So, you are not sure?
You have also opened a PR.
I asked if you have tested it.  But, you have not responded.

**Reply by Cyxapuk-git** [reply]
I have a lot of work, I'm sorry. 
I have tested it on my daughter's phone and it have been working for a week.

**Reply by mtumashenko** [reply]
> I have a lot of work, I'm sorry. I have tested it on my daughter's phone and it have been working for a week.

Can you release an apk in your repo? I can test with an iCan Asia app.

**Reply by Navid200** [reply]
@mtumashenko I can do that, Please give me a few hours.

**Reply by Navid200** [reply]
@mtumashenko This is with iCan Health added.
But, please note this is for testing only.  You will not be able to use Google Drive for transferring to and from.  You will have to use our legacy backup (https://navid200.github.io/xDrip/docs/Backup-Database.html) for that.  

https://drive.google.com/file/d/1APH_yvDVN1P1I-DV-RLazYRpIuj7x-fW/view?usp=sharing 

If you can test, please let us know if it works.  if it does, please provide a screenshot of the iCan Health app main screen as well as the xDrip main screen showing the same readings.

**Reply by mtumashenko** [reply]
> If you can test, please let us know if it works. if it does, please provide a screenshot of the iCan Health app main screen as well as the xDrip main screen showing the same readings.

Thanks alot! It works! Skipped a couple of readings, but mostly works.

xDrip
![image](https://github.com/user-attachments/assets/8821f02e-8198-4fa3-9d56-acdf1ba48118)

iCan
<img width="257" alt="image" src="https://github.com/user-attachments/assets/e0f797af-2110-439d-92ff-a8ae770579c2" />

**Reply by Navid200** [reply]
Would you mind providing two screenshots that show the same readings not one showing 5.5 and the other 6.2?

**Reply by mtumashenko** [reply]
Sorry, my bad, was looking for missed readings and took screenshot of one of them.

Hope this helps!

Notification bar
![image](https://github.com/user-attachments/assets/d0cfb53c-ac0c-434d-ac7b-77b8e107af79)

iCan
![image](https://github.com/user-attachments/assets/5b78334b-d63e-4115-a700-f0d71be8e1d5)

xDrip
![image](https://github.com/user-attachments/assets/574e2739-b5cb-441f-9a0f-83fd6e2a8201)

**Reply by Navid200** [reply]
Great!  Thanks

**Reply by mtumashenko** [reply]
Just wanted to say that all works fine with latest build [xDrip-plus-20250425-7d712fa.apk](https://github.com/NightscoutFoundation/xDrip/releases/download/2025.04.25/xDrip-plus-20250425-7d712fa.apk)

**Reply by warid91** [reply]
> Just wanted to say that all works fine with latest build [xDrip-plus-20250425-7d712fa.apk](https://github.com/NightscoutFoundation/xDrip/releases/download/2025.04.25/xDrip-plus-20250425-7d712fa.apk)

can you help, how did you connect sinocar ican cgm to xdrip app pleases?

**Reply by Navid200** [reply]
@warid91 Try this:  https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html
